.circleci/config.yml

```diff
@@ -119,29 +133,27 @@ jobs:
             echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
             docker push $IMAGE_NAME:latest-bridge
   lint:
-        description: |
-            Check linting with Clippy and rustfmt.
-        resource_class: xlarge
-        executor:
-            name: rust/default
-            tag: 1.71.1
-        environment:
-            RUSTFLAGS: '-D warnings'
-            RUST_LOG: 'debug'
-        steps:
-            - checkout
-            - install-depends
-            - run:
-                name: Run rustfmt
-                command: cargo fmt --all -- --check
-            - run:
-                name: Install Clippy
-                command: rustup component add clippy
-            - setup-and-restore-sccache-cache
-            - run:
-                name: Run Clippy
-                command: cargo clippy --all --all-targets --all-features --no-deps -- --deny warnings
-            - save-sccache-cache
+    description: |
+      Check linting with Clippy and rustfmt.
+    resource_class: xlarge
+    executor: docker-nightly
+    environment:
+      RUSTFLAGS: "-D warnings"
+      RUST_LOG: "debug"
+    steps:
+      - checkout
+      - install-depends-nightly
+      - run:
+          name: Run rustfmt
+          command: cargo +nightly fmt --all -- --check
+      - run:
+          name: Install Clippy
+          command: rustup component add clippy
+      - setup-and-restore-sccache-cache
+      - run:
+          name: Run Clippy
+          command: cargo clippy --all --all-targets --all-features --no-deps -- --deny warnings
```

Comment:
Looking at this now, I assume here `clippy` is running with the nightly version, but because we don't pin Trin to nightly there may be some discrepancies between nightly clippy and stable clippy in the future.

It would be better if we split the `lint` job to `cargo-fmt` and `cargo-clippy` and run clippy with the default stable toolchain.

 If you don't feel like doing it in this PR, we can leave it for a future fix (this will not be a blocker for merging this PR).

Reply:
I could try it in this PR, seems like a simple change

Reply:
all done!